src/shared/components/MSKTabs/MSKTabs.tsx

```diff
@@ -102,25 +154,172 @@ export class MSKTabs extends React.Component<IMSKTabsProps, IMSKTabsState> {
                 effectiveActiveTab = children[0].props.id;
             }
 
-            return <div id={(this.props.id) ? this.props.id : ''}
-                        className={ classnames(this.props.className) }>
-                <ul className="nav nav-tabs">{
-                    React.Children.map(children, (tab: React.ReactElement<IMSKTabProps>) => {
-                        if (!tab || tab.props.hide || (tab.props.loading && (effectiveActiveTab !== tab.props.id))) {
-                            return;
-                        }
-                        let activeClass = (effectiveActiveTab === tab.props.id) ? 'active' : '';
-                        return <li style={{ cursor:'pointer' }} className={activeClass}>
-                            <a onClick={this.setActiveTab.bind(this,tab.props.id)}>{tab.props.linkText}</a>
-                        </li>
-                    })
-                }</ul>
-                <div className="tab-content">{arr}</div>
-            </div>;
+            return (
+                <div
+                    id={(this.props.id) ? this.props.id : ''}
+                    className={ classnames(this.props.className) }
+                >
+                    {this.navTabs(children, effectiveActiveTab)}
+                    <div className="tab-content">{arr}</div>
+                    <ReactResizeDetector handleWidth={true} onResize={this.initOnResize.bind(this)()} />
+                </div>
+            );
         } else {
             return null;
         }
     }
 
-}
+    protected navTabs(children: React.ReactElement<IMSKTabProps>[], effectiveActiveTab: string)
+    {
+        // restart the tab refs before each tab rendering
+        this.tabRefs = [];
+
+        // if pagination is disabled, pages.length and pagesCount will be always 1
+        const pages = this.tabPages(children, effectiveActiveTab);
+        const pageCount = this.state.pageBreaks.length + 1;
+
+        // we need a little style tweak to prevent initial overflow flashing when paging enabled
+        const navBarStyle = this.props.enablePagination ? {
+            maxHeight: 40, border: 0, overflow: "hidden"
```

Comment:
`maxHeight` tweak causes inconsistency when the component is integrated in `cBioPortal` main app.